spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/LazyUtil.java

```diff
@@ -90,6 +90,36 @@ public static boolean isLazyAndNotLoaded(Object object) {
     return false;
   }
 
+  /**
+   * Check if the object is a lazy loaded proxy.
+   *
+   * @param object an object
+   * @return true if the object is a proxy
+   */
+  static boolean isLazy(Object object) {
```

Comment:
What happens if we reuse `isLazyAndNotLoaded` instead? Theoretically, if the object already got loaded, it should have all its data available.